### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by cnasikas**

### CODE REVIEW

File: api/auth/github.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ const strategy = new GitHubStrategy(
         where: {githubId: profile.id},
         defaults: {
           username: profile.username,
-          email: profile.emails[0].value,
+          // email: profile.emails[0].value,
```
Comment (dionyziz) [inline api/auth/github.js]: Check if email exists and add if it does.

File: api/index.js [hunk 2]
```diff
@@ -7,16 +7,25 @@ const bodyParser = require('body-parser')
 const routes = require('./routes')
 const passport = require('./auth')
 const app = express()
+const helmet = require('helmet')
+const cors = require('cors')
 const session = require('express-session');
 
 (async () => {
+  app.use(helmet())
+  app.use(cors({credentials: true, origin: 'http://localhost:3001', methods: ['GET', 'PUT', 'POST']}))
```
Comment (dionyziz) [inline api/index.js]: Separate application URL to .env

File: app/src/assets/css/App.css [hunk 3]
```diff
@@ -21,3 +22,16 @@ body {
...
+  color: #fff;
+}
+
+.assignments {
+  width: 100%;
+}
+
+.page-title {
+  margin: 0;
+}
```
Comment (dionyziz) [inline app/src/assets/css/App.css]: Please accompany visual changes with before/after screenshot

File: app/src/components/List/ListItem.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
```
Comment (dionyziz) [inline app/src/components/List/ListItem.js]: This should be in the assignments subfolder, as it's assignment-specific.

File: app/src/views/Login/Login.js [hunk 5]
```diff
@@ -1,9 +1,10 @@
 import React from 'react'
+import { Redirect } from 'react-router'
 
 export default class Login extends React.Component {
   render () {
     return (
-      <div />
+      <Redirect to='http://localhost:3000/api/auth/github' />
```
Comment (dionyziz) [inline app/src/views/Login/Login.js]: Make URL configurable

File: app/src/components/List/ListItem.js (hunk 4 above)
Comment (cnasikas) [reply app/src/components/List/ListItem.js]: It can be used from lectures, or any view that wants to iterate a list

Comment (dionyziz) [reply app/src/components/List/ListItem.js]: But the links are to assignments.

Comment (cnasikas) [reply app/src/components/List/ListItem.js]: You are right. Missed that. So move it to assignments or take the whole link url as prop ?

Comment (dionyziz) [reply app/src/components/List/ListItem.js]: Either or :)

Comment (cnasikas) [reply app/src/components/List/ListItem.js]: :)

File: app/src/assets/css/App.css (hunk 3 above)
Comment (cnasikas) [reply app/src/assets/css/App.css]: screen shot 2018-05-24 at 16 08 04

**Pull request merged**